validator/project_validator.go

```diff
@@ -1350,10 +1350,11 @@ func validateTVDependsOnTV(source, target model.TVPair, tvToTaskUnit map[model.T
 			source.TaskName, source.Variant)
 	}
 
+	// patches and mainline builds shouldn't depend on anything that's git tag only
```

Comment:
If we aren't allowed to have a dependency with git tag only, we shouldn't actually be checking it here because it will only apply to task sync operations (L1303). This function can only answer whether task A depends on task B in the dependency graph, not whether the dependency graph contains any task with a dependency on a `git_tag_only` task.

Instead, there should be a separate validation function that the dependency graph cannot contain any task that depends on a task with `git_tag_only: true` (maybe in `checkDependencyGraph` since that does a DFS through the whole dependency tree).